### PR TITLE
Framework initial work.

### PR DESCRIPTION
**Pull request by awels**
- Added Framework struct
- Added exec_util from k8s e2e testing framework to allow execution of
  commands in pods.
- Added PVC state checking function
- Added function to check if PVC is empty.
- Added negative test to verify that if a regular PVC is create the importer
  does not do anything to it.

Signed-off-by: Alexander Wels <awels@redhat.com>

### CODE REVIEW

**Comment by awels** [comment]
@jeffvance @davidvossel @copejon Please review, hopefully it makes sense.

File: glide.yaml [hunk 1]
```diff
@@ -17,3 +17,4 @@ import:
   version: 13fc8573324cd2af42c88043b5a7c6e03bf39f95
   repo: https://github.com/kubevirt/qe-tools.git
 - package: github.com/spf13/pflag
+- package: github.com/docker/spdystream
```
Comment (davidvossel) [inline glide.yaml]: what is this dependency used for? I don't see it imported anywhere in our code.

Comment (awels) [reply glide.yaml]: exec_util.go uses it to open the stream to the container.

Comment (davidvossel) [reply glide.yaml]: we aren't importing the docker package anywhere in our code though. This package is only used within the k8s package that's responsible for making the remote connection.

With glide, we don't manage our dependencies' dependencies. That's supposed to be taken care of for us.

try removing this dependency from glide.yaml and running ```glide cc && glide update --strip-vendor``` again. I'd hope that docker dependency get's pulled in. That works in KubeVirt. We have to do the same thing.

**Comment by jeffvance** [comment]
Email sent.
Also, why all the docker spdy files?

**Comment by awels** [comment]
They got pulled in by the streamer from the container.

File: tests/import_test.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+var _ = Describe(testSuiteName, func() {
+	f := tests.NewFramework(namespacePrefix)
+
+	Describe("Creating standard PVC should not trigger importer", func() {
+		It("PVC should remain empty after being bound", func() {
+			pvc := f.CreatePVCFromDefinition(f.GetNamespace().GetName(), tests.NewPVCDefinition("no-import", "1G", nil))
+			// PVC will be pending until bound to a pod, this test will fail if bound by a pod.
+			// NOTE: not 100% sure if this will work with non local volume storage, those might get bound without having
```
[the remark this answers is not in the document]
Comment (awels) [reply tests/import_test.go]: With local-volume PVCs its a little funky. They stay pending until they are attached to a POD, which is where my confusion comes from. But regular PVCs get bound to the PV without a pod getting attached.

**Comment by awels** [comment]
Abandoning in favor of #354

**Pull request closed without merging**